src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,193 @@
+package duke;
+
+import java.io.IOException;
+import java.util.ArrayList;
+
+import javafx.application.Application;
+import javafx.scene.Scene;
+import javafx.scene.control.Button;
+import javafx.scene.control.Label;
+import javafx.scene.control.ScrollPane;
+import javafx.scene.control.TextField;
+import javafx.scene.image.Image;
+import javafx.scene.image.ImageView;
+import javafx.scene.layout.AnchorPane;
+import javafx.scene.layout.Region;
+import javafx.scene.layout.VBox;
+import javafx.stage.Stage;
+
+
+public class Duke extends Application {
+
+    private Image user = new Image(this.getClass().getResourceAsStream("/images/DaUser.png"));
+    private Image duke = new Image(this.getClass().getResourceAsStream("/images/DaDuke.png"));
+
+    private ScrollPane scrollPane;
+    private VBox dialogContainer;
+    private TextField userInput;
+    private Button sendButton;
+    private Scene scene;
+
+    private Storage storage;
+    private TaskList tasks;
+    private Ui ui;
+
+    public Duke() {}
+    /**
+     * Class Constructor specifying filepath.
+     * @param filepath
+     */
+    public Duke(String filepath) {
+        this.ui = new Ui();
+        this.storage = new Storage(filepath);
+        try {
+            tasks = new TaskList(storage.load());
+        } catch (IOException | ClassNotFoundException e) {
+            System.out.println(e);
+            tasks = new TaskList();
+        }
+    }
+
+    /**
+     * Runs the main logic of Duke.
+     * @throws DukeException
+     * @throws IOException
+     * @throws ClassNotFoundException
+     */
+    public void run() throws DukeException, IOException, ClassNotFoundException {
+        ui.start(storage);
+    }
+
+    @Override
+    public void start(Stage stage) {
+        //Step 1. Setting up required components
```

Comment:
This method is a little too long. Maybe you would like to abstract some of it out?